### ISSUE TITLE
Proposal to let `--envdir` load environment variables even for the first worker

### ISSUE THREAD

**Issue by aeroastro** [kazuho/p5-Server-Starter#50]
@kazuho @limitusus

We have been using `--envdir` option introduced in #11.

This feature is really great since we can reload environment variables
just by sending `SIGHUP` or waiting auto restart.

However, I have found a minor pitfall regarding this option during development
and production operations.

I am going to make a patch if you do not see problems on this proposal.

## Current Situation

When `envdir` command is omitted in script to launch `Server::Starter`,
environment variables are not applied to 1st generation worker. Only workers
after 1st generation can read environment variables.
This reload can be invoked deliberately (`SIGHUP`) or not deliberately (auto restart).

IMHO, this is a little bit surprising behavior, and in our production environment,
actually, this is becoming a pitfall.

c.f.
https://github.com/kazuho/p5-Server-Starter/blob/296ee4e45fe984d210f28ac6ca632b2b2040b053/t/07-envdir.t#L53-L55

## Proposal

As an option to resolve this issue, I would like to let `--envdir` to load
environment variables even for the initial worker.

Users those who use `envdir` in run script will not be affected by this change
since load by `--envdir` is executed just after `Server::Starter` is launched,
in other words, just after `envdir` is executed.
On the other hand, users those who do NOT use `envdir` can avoid the issue
described above.

This feature can loose coupling between `Server::Starter` and `daemontools`,
and make `--envdir` a portable great option inspired from `daemontools`.

**Comment by limitusus** [comment]
I have no objection against the idea.
The first worker's loading envvars is more intuitive to me.

**Comment by aeroastro** [comment]
Thank you! I will make some patch on this.
